libp2p/builders.nim

```diff
@@ -0,0 +1,198 @@
+import
+  options, tables, chronos, bearssl,
+  switch, peerid, peerinfo, stream/connection, multiaddress,
+  crypto/crypto, transports/[transport, tcptransport],
+  muxers/[muxer, mplex/mplex],
+  protocols/[identify, secure/secure, secure/noise],
+  connmanager, upgrademngrs/muxedupgrade
+
+export
+  switch, peerid, peerinfo, connection, multiaddress, crypto
+
+type
+  SecureProtocol* {.pure.} = enum
+    Noise,
+    Secio {.deprecated.}
+
+  TcpTransportOpts = object
+    enable: bool
+    flags: set[ServerFlags]
+
+  MplexOpts = object
+    enable: bool
+    newMuxer: MuxerConstructor
+
+  SwitchBuilder* = ref object
+    privKey: Option[PrivateKey]
+    address: MultiAddress
+    secureManagers: seq[SecureProtocol]
+    mplexOpts: MplexOpts
+    tcpTransportOpts: TcpTransportOpts
+    rng: ref BrHmacDrbgContext
+    inTimeout: Duration
+    outTimeout: Duration
+    maxConnections: int
+    maxIn: int
+    maxOut: int
+    maxConnsPerPeer: int
+    protoVersion: string
+    agentVersion: string
+
+proc init*(T: type[SwitchBuilder]): T =
```

Comment:
`new` because it's a `ref object`

Reply:
I'll change it. But as a note, there is an issue with paramless `new`, I believe I ran into it and nim wasn't able to tell it apart from the system `new`. If I can repro it, I'll log it as an issue in the style guide.